### PR TITLE
Add color palette edit functionality to global styles

### PR DESCRIPTION
**Pull request by jorgefilipecosta**
This PR allows the user to configure a global color palette or even a color palette for a specific block.

## Limitations
- The colors created by the user don't work well when applied, that happens because we are not automatically generating CSS classes for theme.json yet and rely on themes providing them. There will be a parallel PR that fixes that.
- We don't show the default colors applied by the theme. That happens because we don't yet retrieve the theme and core defaults on global styles, e.g: if the theme sets a default font size the global styles sidebar does not show it. That is an existing issue with global styles that is going to be solved in parallel.
- When we change the color palette, and then go the block editor on the edit site, the new colors don't appear, they only appear after the saving and reload of the editor. That happens because the settings passed to the block editor are a static object retrieved from the server, it should be dynamic as the user can now change these settings. Another PR will solve this issue.
- The UI may take advantage of some enhancements. @noahshrader  I tried to follow your mockups adapting them to the current state of the sidebar, but it still required lots of custom code because we did not have any component that fits. I am open to feedback for changes that should happen on the first version.

## Screenshots 

image

image

image

## Tests
I added a custom color palette I saved the design and verified the colors were persisted.
I verified I could change the color code, name, and slug of existing colors.
I verified I could remove colors.

### CODE REVIEW

**Comment by aristath** [comment]
Just tested this one :+1: 
Additional limitation: If all custom colors are deleted, on save & refresh it reverts to the default colors instead of the theme.json colors

**Comment by jorgefilipecosta** [comment]
> Additional limitation: If all custom colors are deleted, on save & refresh it reverts to the default colors instead of the theme.json colors

There is an important related point: the state of having no user color palette (using the default/theme colors) and the state of the user intentionally setting an empty color palette are different. I added a button to allow the user to remove the user color palette and get back to defaults after removing all the colors.
@aristath I'm not being able to reproduce the issue. After I'm back to defaults the defaults are the theme and the core colors (if theme colors don't exist). I guess the issue may have been solved as a side effect of another change.

File: packages/components/src/color-edit/index.js [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+}
+
+function ColorOption( {
+	color,
+	name,
+	slug,
+	onChange,
+	onRemove,
+	onConfirm,
+	confirmLabel = __( 'Ok' ),
```
Comment (mcsf) [inline packages/components/src/color-edit/index.js:40]:
```suggestion
	confirmLabel = __( 'OK' ),
```

File: packages/components/src/color-edit/index.js [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+		>
+			<div className="components-color-edit__color-option-main-area">
+				<Dropdown
+					renderToggle={ ( { isOpen, onToggle } ) => (
+						<>
+							<DropdownOpenOnMount
+								shouldOpen={ isEditingColorOnMount }
+								isOpen={ isOpen }
+								onToggle={ onToggle }
+							/>
```
Comment (mcsf) [inline packages/components/src/color-edit/index.js:84]: Interesting :)

**Review by mcsf — APPROVED** [review]
Feels nice! Looking forward for the subsequent PRs for the rest of the system.

**Pull request merged**